architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,693 @@
+# Radius BicepDeployment Controller
+
+* **Author**: Will Smith (@willdavsmith)
+
+## Overview
+
+<!--
+Provide a succinct high-level description of the component or feature and 
+where/how it fits in the big picture. The overview should be one to three 
+paragraphs long and should be understandable by someone outside the Radius
+team. Do not provide the design details in this, section - there is a
+dedicated section for that later in the document.
+-->
+
+Today, users of Radius and future adopters of Radius use many Kubernetes-specific tools in their production workflows. Most of these tools operate on Kubernetes resources exclusively - which presents a problem when trying to deploy resources defined in Bicep manifests. This design proposes the creation of a new Kubernetes reconciler in Radius called the BicepDeployment Reconciler that will allow users to deploy resources defined in Bicep manifests using Kubernetes tooling.
```

Comment:
In the future, we will want to support ways to model applications other than Bicep. I'm guessing that you are assuming that Radius will always translate an application definition into Bicep. Is that the assumption? Should this be more of a Radius deployment controller?

Reply:
This is true, but we'll have different ways of handling different IAC techniques. This design is specific to Bicep.